### ISSUE TITLE
excessive use of intermediate nodes in French mappings

### ISSUE THREAD

**Issue by VladimirAlexiev** [dbpedia/mappings-tracker#8]
Many French mappings use intermediate nodes excessively.

Eg http://mappings.dbpedia.org/index.php/Mapping_fr:Infobox_Ville_de_Serbie would produce:

```
<place>
  takePlace [a Place; name "nom division1"; sharingOut <division1>];
  takePlace [a Place; name "nom division2"; sharingOut <division2>];
  politicalLeader [a PoliticalFunction; 
     mayor <mayor>; activeYearsStartYear; activeYearsEndYear]
  demographics [a Demographics; populationTotal 123; year "2014"]
  wholeArea [a Area; value 12345].
```
- I have no clue what takePlace and sharingOut mean, but I think takePlace should become hasPart and sharingOut should become rdf:type (if it's appropriate).
- politicalLeader is used correctly, and used by other language mappings
- demographics (and demographicsAsOf) and year are not useful, since everyone else does it like this.

```
      <place> populationTotal 123; populationAsOf "2014".
```

There are previousDemographics and agglomerationDemographics that could be useful,but these are not yet used
- Similarly for wholeArea and value: everyone else does it like this:

```
      <place> areaLand 12345.
```

**Comment by VladimirAlexiev** [comment]
Did they just make up takePlace, sharingOut, demographics, demographicsAsOf ????

**Comment by jplu** [comment]
Hi,

The property `takePlace` can be indeed replaced by something like `isPartOf` yes.

The property `sharingOut` is there to define the divisions (province, district, etc.) where the city belongs so it's not a type. It's an intermediate node mainly to associate the name of those divisions to the division itself. But I'm agree that the name is not very representative of what it represents. Let's take this example :

```
{{Infobox Ville de Serbie
| nom de division          = Province
| division                 = [[Serbie centrale]]
| nom de division2         = District
| division2                = [[Belgrade (district)|Belgrade]]
}}
```

You can see that there is strings and internal links to Wikipedia. If you don't use a intermediate node, how do you know that `Province` is associated to the resource represented by `[[Serbie centrale]]` and not `[[Belgrade (district)|Belgrade]]` ?

'The property `demographics` is there to associate a date to a population. Because you can get many population, one per year (2011, 2012, 2013, etc.) for a populated place.

For the others properties you mentioned it's the same explanation than for `demographics`, they are there to associate a year to a number. You can use the previous example and extends it to those properties for a representation of the problem.

Hope my explanation are clear, in any case don't hesitate to ask.

**Comment by VladimirAlexiev** [comment]
- Province is an attribute of Serbie centrale, not of the city, and District is an attribute of Belgrade (district), not of the city. Therefore "nom de division" should be skipped from the mapping, it's redundant information
- 2011, 2012, 2013: Please give an actual example from wikipedia
- How about wholeArea & value

**Comment by jplu** [comment]
Agree they are not associated to the city, but I gonna push the example deeper :

```
{{Infobox Ville de Serbie
| nom de division          = Province
| division                 = Serbie centrale
| nom de division2         = District
| division2                = Belgrade
}}
```

Now there is only strings, so you cannot go through the links to get what kind of division is "Belgrade".

For `wholeArea` some infoboxes models associate also others properties to an Area than just the value. I don't have examples for this and the population in mind right now sorry, but I know when I was studying the french Wikipedia dump I met many of those cases.

**Comment by VladimirAlexiev** [comment]
> there is only strings, so you cannot go through the links to get what kind of division is "Belgrade".

Now I understand.
- But why call it "sharingOut" instead of dc:type, which has been defined 20 years ago and is widely used? I've now defined it in the wiki so it can be used: http://mappings.dbpedia.org/index.php/OntologyProperty:Dc:type. Please note the comment: we need documentation like this for every prop & class!
- instead of "takePlace", use "part", eg as in http://mappings.dbpedia.org/index.php/Mapping_ru:Административная_единица. 

So the example should become:

```
<place>
    part [a Place; foaf:name "nom division1"; dc:type <division1>]
```

I've checked in the downloadable ontology file, and there are 2 more:

```
:sharingOut a owl:DatatypeProperty , rdf:Property ;
    rdfs:label "sharing out"@en ;
    rdfs:domain :PopulatedPlace ;
    rdfs:range xsd:string ;
:sharingOutArea a owl:DatatypeProperty , rdf:Property ;
    rdfs:label "sharing out area"@en ;
    rdfs:domain :PopulatedPlace ;
    rdfs:range xsd:string ;
:sharingOutName a owl:ObjectProperty , rdf:Property ;
    rdfs:label "sharing out name of a settlement"@en ;
    rdfs:domain :Settlement ;
    rdfs:range :PopulatedPlace ;
```

Can you please show an example in Turtle of potential data using these properties? (Like the example in my first comment).

sharingOutArea and sharingOutName are not used, except in http://mappings.dbpedia.org/index.php/Mapping_eo:Geokesto: libera1_tipo -> sharingOutName. "libera typo" means "free type", so I think they mean a string, and because sharingOutName is an objectProperty they'll get nothing. They should use "type" (if the target is a page), or "dc:type" (if the target is a string).

> wholeArea some infoboxes models associate also others properties to an Area than just the value. I don't have examples for this and the population in mind right now sorry

I checked at http://fr.dbpedia.org/sparql:

```
  select * {[] dbpedia-owl:wholeArea [?p ?x] filter (?p!=rdf:type && ?p!=dbpedia-owl:value)}
```

Indeed, some use min, max, rank.
- I've commented http://mappings.dbpedia.org/index.php/OntologyClass:Area, see if you agree
- it'd be better to use a new data prop "sortOrder" of type xsd:integer, else you get "12"@fr as you see above (and rank="12" is really a different thing from rank="Marechal"). But there are [many uses of rank](http://mappings.dbpedia.org/index.php?title=Special:Search&ns0=1&ns200=1&ns202=1&ns204=1&ns206=1&ns208=1&ns210=1&ns212=1&ns214=1&ns216=1&ns218=1&ns220=1&ns222=1&ns224=1&ns226=1&ns228=1&ns230=1&ns232=1&ns236=1&ns238=1&ns240=1&ns242=1&ns246=1&ns250=1&ns254=1&ns256=1&ns262=1&ns264=1&ns268=1&ns272=1&ns274=1&ns282=1&ns284=1&ns304=1&ns312=1&ns328=1&ns378=1&ns396=1&redirs=1&search=%22ontologyproperty%3Drank%22&limit=50&offset=0) already, don't know who can fix them...

Now please research the uses of, and add an appropriate comment to the class _Demographics_.

**Comment by jplu** [comment]
> But why call it "sharingOut" instead of dc:type, which has been defined 20 years ago and is widely used?

You have totally right, it is not the good name.

> They should use "type"

You mean this property ? http://mappings.dbpedia.org/index.php/OntologyProperty:Type

> sharingOutArea and sharingOutName are not used

Ideed, they are not used, so we can delete them. And sorry I don't remember why I have created those ones.

> I've commented http://mappings.dbpedia.org/index.php/OntologyClass:Area, see if you agree

Very good yes.

> it'd be better to use a new data prop "sortOrder" of type xsd:integer

I am totally agree, yes.

> Now please research the uses of

Once I remember yes I will give example.

> add an appropriate comment to the class Demographics.

Is-it ok for you ? http://mappings.dbpedia.org/index.php/OntologyClass:Demographics

**Comment by VladimirAlexiev** [comment]
You don't need to remember, use SPARQL. (I first found the 3 most often used props, then discovered the remaining one "rank")

```
select * {?x a dbpedia-owl:Demographics; ?y ?z 
  filter (!(?y in (rdf:type, dbpedia-owl:year, dbpedia-owl:populationTotal)))} limit 200
```

I've edited the description to be more specific, please take a look

**Comment by VladimirAlexiev** [comment]
Yes, when I don't give a prefix I mean dbo:. I.e. by "type" I mean "dbo:type" or exactly the prop that you have given a link for, above

**Comment by jplu** [comment]
After playing a bit with SPARQL I didn't found any usage of those properties, so we can certainly remove them and modify the mapping for something better.

> I've edited the description to be more specific, please take a look

It's ok !

**Comment by VladimirAlexiev** [comment]
Which are "those properties"?

Sorry, this issue has gotten too big :-( Probably best to split it

**Comment by jplu** [comment]
Sorry by "those properties" I mean "sharingOutArea" and "sharingOutName".

**Comment by VladimirAlexiev** [comment]
If unused, definitely kill them

**Comment by jplu** [comment]
Done !

**Comment by VladimirAlexiev** [comment]
Instead of takePlace, you should use geographic structure properties with reasonable names and that are used by others. 
I have not researched this question and AFAIK nobody has defined such best practice: maybe you can do that?
- eg http://mappings.dbpedia.org/index.php/Mapping_de:Infobox_Gemeinde_in_Spanien: uses region, province, locatedInArea
- region is used in over 200 mappings: http://mappings.dbpedia.org/index.php?title=Special:Search&ns0=1&ns200=1&ns202=1&ns204=1&ns206=1&ns208=1&ns210=1&ns212=1&ns214=1&ns216=1&ns218=1&ns220=1&ns222=1&ns224=1&ns226=1&ns228=1&ns230=1&ns232=1&ns236=1&ns238=1&ns240=1&ns242=1&ns246=1&ns250=1&ns254=1&ns256=1&ns262=1&ns264=1&ns268=1&ns272=1&ns274=1&ns282=1&ns284=1&ns304=1&ns312=1&ns328=1&ns378=1&ns396=1&redirs=0&search=region&limit=100&offset=0
- etc etc
